### DISCUSSION TITLE
Paintings

### DISCUSSION THREAD

**Discussion by JamjamSA** [APDevTeam/Movecraft#600] [Q&A] [answered]
So basically, I made a ship that's quite huge (which doesn't really matter) and for some reason when I try rotating, the paintings always seem to come off. When I'm moving forward, backward left and/or right, it's fine but when turning my ship around, the paintings come off. Is there an option for that in the .craft file that I don't know about?

**Comment by Thenolypus** [comment, marked as answer]
From what I understand, paintings are considered entities. ~~and I assume moving entities along with the craft is a feature that is not yet fully implemented?~~ ~~Or it could be a bug waiting to be fixed for 1.20.1 (which was the version I tested on). It is either that the feature to move entities does exist, just hidden deep somewhere, or it is still in development. I did go through the source code a little bit, I do see some hints of codes about entities in certain scripts (although my java is very rusty), but I could not come to an answer. This issue (#508) that was implemented on one of the source code was what I found, but was not able to determine to what extent is it implemented.~~

~~Interesting note during my testing however: I tried moving a cow on the craft. When it was stationary, it did not move along with the craft, however as it started (wandering via pathfinding) moving around, it went along with the craft as it moved. Once the cow stopped moving, viola, the craft forgets its existence.~~

EDIT: I have edited the message above and cancel out irrelevant statements that I have made earlier. I did a little further messing around in the .craft files and figured out how to help you. I looked into the source code in [CraftType.java](https://github.com/APDevTeam/Movecraft/blob/61aa121df441bd2e4715f2c45a6a558349b14407/modules/api/src/main/java/net/countercraft/movecraft/craft/type/CraftType.java#L484) and found two lines of code that was of interest in line 483 & 484 respectively:
`registerProperty(new BooleanProperty("moveEntities", MOVE_ENTITIES, type -> true));`
`registerProperty(new BooleanProperty("onlyMovePlayers", ONLY_MOVE_PLAYERS, type -> true));`

This shows that in each craft type you create, without specifying these variables, they will default to true. `moveEntities` would move entities along with the craft, however it will conflict with `onlyMovePlayers`.

The fix to this?
Assuming you are running a Minecraft server, navigate to your CraftType files in plugins -> Movecraft -> types -> CraftType.craft. Add these two lines into your CraftType.craft (e.g. Airship.craft if you are using Airship) file:
`onlyMovePlayers: false`
`moveEntities: true`

I fitted these two lines above the `moveblocks:` entry, tested it out with a cow and a painting after that, now it works! Hope this helps!
